internal/handlers/common/update.go

```diff
@@ -77,6 +77,26 @@ func UpdateDocument(doc, update *types.Document) (bool, error) {
 			for _, incKey := range incDoc.Keys() {
 				incValue := must.NotFail(incDoc.Get(incKey))
 
+				if strings.Contains(incKey, ".") {
+					path := types.NewPathFromString(incKey)
+					innerValue, err := doc.GetByPath(path)
+					if err != nil {
+						err = doc.InsertByPath(path, incValue)
+						if err != nil {
+							return false, err
+						}
+						changed = true
+						continue
+					}
+
+					incremented, err := addNumbers(incValue, innerValue)
```

Comment:
I think we will have the same issue with $dec.

Let's see if we have the same issue with `$set` (when we replace the value with the same value). If yes, I would say that code should be generalized

Reply:
I think it could be refactored when I would add `$set` support. For now, I suggest leaving it as is. There is no `$dec` operator, btw :smiley: